src/sentinel1_reader/sentinel1_reader.py

```diff
@@ -0,0 +1,418 @@
+
+        # choose nearest azimuth FM rate
+        d_seconds = 0.5 * (n_lines - 1) * azimuth_time_interval
+        sensing_mid = sensing_start + datetime.timedelta(seconds=d_seconds)
+        az_fm_rate = get_nearest_polynomial(sensing_mid, az_fm_rate_list)
+
+        # choose nearest doppler
+        poly1d = get_nearest_polynomial(sensing_mid, doppler_list)
+        lut2d = doppler_poly1d_to_lut2d(poly1d, starting_range,
+                                        range_pxl_spacing, (n_lines, n_samples),
+                                        azimuth_time_interval)
+        lut1d = isce3.core.avg_lut2d_to_lut1d(lut2d)
+        doppler = Doppler(poly1d, lut2d)
+
+        # get orbit from state vector list/element tree
+        sensing_duration = datetime.timedelta(
+            seconds=n_samples * azimuth_time_interval)
+        orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
+                                osv_list)
+
+        # determine burst offset and dimensions
+        # TODO move to own method
+        first_valid_samples = [int(val) for val in burst_list_element.find('firstValidSample').text.split()]
+        last_valid_samples = [int(val) for val in burst_list_element.find('lastValidSample').text.split()]
+
+        first_valid_line = [x > 0 for x in first_valid_samples].index(True)
+        n_valid_lines = [x > 0 for x in first_valid_samples].count(True)
+        last_line = first_valid_line + n_valid_lines - 1
+
+        first_valid_sample = max(first_valid_samples[first_valid_line],
+                                 first_valid_samples[last_line])
+        last_sample = min(last_valid_samples[first_valid_line],
+                          last_valid_samples[last_line])
+        n_valid_samples = last_sample - first_valid_sample
+
+        burst_id = f't{track_number}_{subswath_id.lower()}_b{id_burst}'
+
+        bursts[i] = Sentinel1BurstSlc(sensing_start, radar_freq, wavelength,
+                                      azimuth_steer_rate, azimuth_time_interval,
+                                      slant_range_time, starting_range,
+                                      range_sampling_rate, range_pxl_spacing,
+                                      (n_lines, n_samples), az_fm_rate, doppler,
+                                      rng_processing_bandwidth, pol, burst_id,
+                                      platform_id, center_pts[i], boundary_pts[i],
+                                      orbit, tiff_path, i, first_valid_sample,
+                                      last_sample, first_valid_line, last_line)
+
+    return bursts
+
+def zip2bursts(zip_path: str, osv_list: ET.Element,
+               n_subswath: int, pol: str):
+    '''
+    Find bursts in a Sentinel 1 zip file
+
+    Parameters:
+    -----------
+    zip_path : str
+        Path the zip file.
+    osv_list : xml.etree.ElementTree.Element
+        ElementTree containing orbit state vectors
```

Comment:
@LiangJYu I find this argument a bit strange. zip2burst is the interface with the user of the reader and it looks very user unfriendly to me to require an argument which is xml.etree.ElementTree.Element!
Is there any reason you ended up with this approach?  can we avoid this. Shall we simply require an orbit file and not an orbit directory so that the user of the reader will be responsible to pass the correct orbit file.

Reply:
Actually I think the reader should just simply take the orbit file and not the orbit directory. As discussed [here](https://github.com/opera-adt/COMPASS/discussions/3) it should be the responsibility of the user of the S1 reader at higher level to find the right orbit file and pass it to the workflows such that the reader can use.

Reply:
I agree with you it's currently not as user friendly. I opted to do this so the same orbit XML doesn't need to be reread in the event we want to process all bursts/all polarizations for a zip. This is slightly faster (<< 1 sec to read the XML) but not significantly so in grand scheme of things.

I like the idea of the reader take in the correct orbit file. We have or will have tools available to help users pick the correct file. The orbit reader can pick the correct file (if it exists) from a directory. The same logic can be fairly easily extended to scrape the ASF orbit site.

Reply:
If the improvement is << 1 sec, then I suggest to not sacrifice the user friendliness. 
Yes, I think the reader should expect the correct orbit file and not orbit directory.